### ISSUE TITLE
[BUG] Exception in LineMarkerSyntax

### ISSUE THREAD

**Issue by jansorg** [arrow-kt/arrow-meta#630]
**Describe the bug**
I edited some sample code and got the exception.

Exception:
```
During querying provider arrow.meta.ide.dsl.editor.lineMarker.LineMarkerSyntax$addLineMarkerProvider$1@79c970cf (class arrow.meta.ide.dsl.editor.lineMarker.LineMarkerSyntax$addLineMarkerProvider$1)

java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.base/java.util.Collections$SingletonList.get(Collections.java:4849)
	at arrow.meta.ide.plugins.proofs.markers.CoercionLineMarkerKt.participatingTypes(coercionLineMarker.kt:52)
	at arrow.meta.ide.plugins.proofs.markers.CoercionLineMarkerKt.participatingTypes(coercionLineMarker.kt:31)
	at arrow.meta.ide.plugins.proofs.markers.CoercionLineMarkerKt.isCoerced(coercionLineMarker.kt:36)
	at arrow.meta.ide.plugins.proofs.markers.ValueArgumentCoercionKt$implicitCoercionValueArgumentLineMarker$1.invoke(valueArgumentCoercion.kt:26)
	at arrow.meta.ide.plugins.proofs.markers.ValueArgumentCoercionKt$implicitCoercionValueArgumentLineMarker$1.invoke(valueArgumentCoercion.kt)
	at arrow.meta.ide.plugins.proofs.markers.ValueArgumentCoercionKt$addLineMarkerProviderM$3.invoke(valueArgumentCoercion.kt:52)
	at arrow.meta.ide.plugins.proofs.markers.ValueArgumentCoercionKt$addLineMarkerProviderM$3.invoke(valueArgumentCoercion.kt)
	at arrow.meta.ide.dsl.editor.lineMarker.LineMarkerSyntax$addLineMarkerProvider$1.getLineMarkerInfo(LineMarkerSyntax.kt:47)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryProviders(LineMarkersPass.java:171)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.lambda$doCollectInformation$3(LineMarkersPass.java:96)
	at com.intellij.codeInsight.daemon.impl.Divider.divideInsideAndOutsideInOneRoot(Divider.java:79)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.doCollectInformation(LineMarkersPass.java:91)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:52)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:442)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:435)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:627)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:572)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:434)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:410)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:168)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:168)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:408)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:171)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**To Reproduce**
- Open the project and file https://github.com/arrow-kt/arrow-typeproofs/blob/e63b03f144993e103c143a0dba27431537223e7f/src/main/kotlin/arrow/proofs/demo/kinds/kinds.kt#L20 
- Append some lines 

**Environment (please complete the following information):**
Current git

 - [Kotlin Intellij IDEA plugin version](https://plugins.jetbrains.com/plugin/6954-kotlin/versions)  1.3.61-SNAPSHOT
- Intellij IDEA version: IC-2020.1

**Additional context**

**Comment by i-walker** [comment]
Thanks for opening the issue,
it was solved in #640 
Please close this if that experience has been dealt with updating to the latest commit in master.

**Comment by i-walker** [comment]
I have to excuse my last message. #640 addressed a different error, related issue, but I was still able to reproduce the error.
